src/riscv/lib/src/jit/builder/outcome_map.rs

```diff
@@ -26,6 +26,20 @@ use std::ops::Range;
 
 use crate::jit::builder::instr_map::InstrId;
 
+/// Probability of taking an outcome of a particular instruction.
+///
+/// As detailed in RISC-V Control Transfer Instructions specification (2.5),
+/// backward-branches should be treated as likely taken, while forward-branches
+/// should be treated as likely not-taken. Also, exception handlers should be treated
+/// as likely not-taken (except for a few instructions, such as `ECall`, which are guaranteed to
+/// result in an exception).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum OutcomeProbability {
+    Guaranteed,
+    High,
+    Low,
+}
```

Comment:
I reckon this is in the wrong module. Probably worth placing it closer to `crate::jit::builder::instruction::Outcome`.